tools.go

```diff
@@ -0,0 +1,11 @@
+//go:build tools
```

Comment:
why do we need this flag? what is the target that sets this flag in the build command?

Reply:
This is a [recommendation from Ginkgo](https://github.com/onsi/ginkgo#go-module-tools-package) so that we can run the `ginkgo` binary from `vendor`. If we just added the import path to `go.mod`, it would be removed as soon as we ran `go mod tidy`. This file prevents that.

Reply:
```
vendor/sigs.k8s.io/controller-runtime/Makefile: cd $(TOOLS_DIR) && go build -tags=tools -o bin/go-apidiff github.com/joelanford/go-apidiff
vendor/sigs.k8s.io/controller-runtime/Makefile: cd $(TOOLS_DIR) && go build -tags=tools -o bin/controller-gen sigs.k8s.io/controller-tools/cmd/controller-gen
```

Reply:
Ideally I would like that the linter will always check everything